src/transactions/test/LiquidityPoolDepositTests.cpp

```diff
@@ -0,0 +1,352 @@
+                ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, -1, 100, Price{1, 1}, Price{1, 1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            // bad maxAmountB
+            REQUIRE_THROWS_AS(
+                root.liquidityPoolDeposit({}, 100, 0, Price{1, 1}, Price{1, 1}),
+                ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, 100, -1, Price{1, 1}, Price{1, 1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            // bad minPrice.n
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, 100, 100, Price{0, 1}, Price{1, 1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, 100, 100, Price{-1, 1}, Price{1, 1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            // bad minPrice.d
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, 100, 100, Price{1, 0}, Price{1, 1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, 100, 100, Price{1, -1}, Price{1, 1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            // bad maxPrice.n
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, 100, 100, Price{1, 1}, Price{0, 1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, 100, 100, Price{1, 1}, Price{-1, 1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            // bad maxPrice.d
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, 100, 100, Price{1, 1}, Price{1, 0}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+            REQUIRE_THROWS_AS(root.liquidityPoolDeposit(
+                                  {}, 100, 100, Price{1, 1}, Price{1, -1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_MALFORMED);
+        });
+    }
+
+    SECTION("both non-native without liabilities")
+    {
+        for_versions_from(18, *app, [&] {
+            root.setOptions(setFlags(AUTH_REQUIRED_FLAG));
+
+            // This section is all about depositing into an empty pool
+            auto a1 = root.create("a1", minBal(10));
+
+            // No trust
+            REQUIRE_THROWS_AS(a1.liquidityPoolDeposit(pool12, 400, 900,
+                                                      Price{1, 1}, Price{1, 1}),
+                              ex_LIQUIDITY_POOL_DEPOSIT_NO_TRUST);
```

Comment:
Nit: Just noticed this - most of the `ex_LIQUIDITY_POOL_DEPOSIT_NO_TRUST` checks in this file are hitting the same condition (pool share trustline is missing). This could be cleaned up, but it's up to you.

Reply:
Fixed.